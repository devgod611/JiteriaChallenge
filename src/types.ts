export type UserListType = {
  users: UserType[];
};

export type UserType = {
  email: string,
  id: number,
  name: string,
  phone: string,
  username: string,
  website: string,
};

export type UserState = {
  users: UserType[];
};

export type RootState = {
  UserState: UserState;
};
