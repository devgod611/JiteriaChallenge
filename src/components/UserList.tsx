import React, { ReactElement } from "react";
import { List, Card } from "antd";
import { EditOutlined, DeleteFilled, HeartOutlined } from "@ant-design/icons";
import CardMeta from "./CardMeta";
import { GET_AVATAR_URL } from "../redux/constants";
import { UserListType } from "../types";

function UserList({ users }: UserListType): ReactElement {
  return (
    <List
      data-id="list"
      style={{ marginTop: 20, overflowY: "auto", overflowX: "hidden" }}
      bordered
      dataSource={users}
      grid={{ gutter: 16, xs: 1, sm: 2, md: 2, lg: 3, xl: 4, xxl: 5 }}
      renderItem={(item) => {
        return (
          <List.Item data-id="list-item">
            <Card
              cover={
                <img
                  alt="example"
                  src={GET_AVATAR_URL(item.username)}
                  className="avatar"
                />
              }
              actions={[
                <HeartOutlined key="like" style={{ color: "red" }} />,
                <EditOutlined key="edit" />,
                <DeleteFilled key="delete" />,
              ]}
            >
              <CardMeta item={item} />
            </Card>
          </List.Item>
        );
      }}
    />
  );
}

export default UserList;
