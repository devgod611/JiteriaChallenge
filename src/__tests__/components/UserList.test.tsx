import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import UserList from "../../components/UserList";
import { UserType } from "../../types";

describe("UserList", () => {
  describe("rendering", () => {
    let wrapper: ShallowWrapper;
    const users: UserType[] = [
      {
        email: "Sincere@april.biz",
        id: 1,
        name: "Leanne Graham",
        phone: "1-770-736-8031 x56442",
        username: "Bret",
        website: "hildegard.org"
      }
    ];
    beforeEach(() => {
      wrapper = shallow(<UserList users={users} />);
    });

    it("renders the component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should have List", () => {
      expect(wrapper.find('[data-id="list"]').length).toEqual(1);
    });
  });
});
