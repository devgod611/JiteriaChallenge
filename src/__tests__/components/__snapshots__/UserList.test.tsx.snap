// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList rendering renders the component 1`] = `
<List
  bordered={true}
  data-id="list"
  dataSource={
    Array [
      Object {
        "email": "Sincere@april.biz",
        "id": 1,
        "name": "Leanne Graham",
        "phone": "1-770-736-8031 x56442",
        "username": "Bret",
        "website": "hildegard.org",
      },
    ]
  }
  grid={
    Object {
      "gutter": 16,
      "lg": 3,
      "md": 2,
      "sm": 2,
      "xl": 4,
      "xs": 1,
      "xxl": 5,
    }
  }
  renderItem={[Function]}
  style={
    Object {
      "marginTop": 20,
      "overflowX": "hidden",
      "overflowY": "auto",
    }
  }
/>
`;
